server/db.go

```diff
@@ -232,6 +233,28 @@ func ExecuteRetryable(fn func() error) error {
 	return nil
 }
 
+// ExecuteRetryablePgConn Retry functions that perform non-transactional database operations on PgConn
+func ExecuteRetryablePgConn(ctx context.Context, db *sql.DB, fn func(conn *pgx.Conn) error) error {
+	c, err := db.Conn(ctx)
+	if err != nil {
+		return err
+	}
+	defer c.Close()
+	return c.Raw(func(dc any) error {
+		conn := dc.(*stdlib.Conn).Conn()
+	retry:
+		if err := fn(conn); err != nil {
+			var pgErr *pgconn.PgError
+			if errors.As(err, &pgErr) && pgErr.Code == pgerrcode.SerializationFailure {
+				// A recognised error type that can be retried.
+				goto retry
```

Comment:
Not a fan of the `goto`. Can we rewrite this more neatly, and also implement the max retries we've done elsewhere already?